### DISCUSSION TITLE
What are people using for crash reporting?

### DISCUSSION THREAD

**Discussion by Marv51** [microsoft/WindowsAppSDK#924] [General]
Hi, 
Before WinUI 3 I was using [AppCenter](https://appcenter.ms/) to gather crash reports and get some basic analytic insights into the usage of my app.

AppCenter does not work with [WinUI 3](https://github.com/microsoft/appcenter-sdk-dotnet/issues/1523). 

So, any recommendations? What are people using for this? 

(My app is sideloaded, so the store infrastructure is not a replacement)

**Comment by riverar** [comment]
This doesn't seem appropriate for this repository or the issues tab. That said, we use [Bugsnag](https://www.bugsnag.com/). Not thrilled about it but they were gracious enough to offer us an open-source project license. We were actually going to switch to AppCenter at some point, ha!

**Comment by michael-hawker** [comment]
@Marv51 looks like you already filed https://github.com/microsoft/appcenter-sdk-dotnet/issues/1523? (Took me a while to find their presence on GitHub from their main site.)

> This doesn't seem appropriate for this repository or the issues tab.

@riverar I don't know if I agree with that. 

I mean asking a question about WinUI 3 and the general ecosystem here doesn't seem like too far of a stretch. The WinUI folks should at least be aware that there's a gap here and they may know who to go contact on the App Center team if they're having trouble with investigating how to support WinUI 3 in the future.

**Edit** - missed the original issue link to App Center in the opening post, and took a while to find the round-about way (they don't seem to link to their GitHubs from any of their main sites).

**Comment by riverar** [comment]
@michael-hawker Fair enough!

**Comment by andrewleader** [comment]
Transferring this to a discussion (excellent question/discussion!)

**Reply by Marv51** [reply]
Thanks!

**Comment by benstevens48** [comment]
I haven't tried it personally, but I believe that for packaged apps you can get some basic info for free without altering your app's code at all. See https://docs.microsoft.com/en-us/windows/win32/appxpkg/windows-desktop-application-program. If it's anything like the Store version then there may be some delay before it starts to give accurate stack traces, and the other analytics are likely not as reliable as AppCenter.

**Comment by mrlacey** [comment]
Has anyone tried using [Application Insights](https://docs.microsoft.com/en-us/azure/azure-monitor/app/windows-desktop) with a WinUI3 app?

**Comment by triplef** [comment]
We recently switched to [Sentry](https://sentry.io) across Win/Mac/iOS/Android and are very happy.

We used both App Center and Firebase Crashlytics before, but the former seems to be no longer maintained and doesn’t support Win32 apps, and the latter doesn’t support Windows at all.
